src/ucp/rndv/rndv_ppln.c

```diff
@@ -199,7 +199,9 @@ void ucp_proto_rndv_ppln_send_frag_complete(ucp_request_t *freq, int send_ack)
 
 static ucs_status_t ucp_proto_rndv_recv_ppln_reset(ucp_request_t *req)
 {
-    ucs_assert(req->send.rndv.ppln.ack_data_size == 0);
+    ucs_assertv(req->send.rndv.ppln.ack_data_size <= 0,
```

Comment:
how can it be < 0? maybe try to avoid it?

Reply:
That's what we have discussed, we subtract `overlap` from this field during the `progress()` so when it sums up during completion, the size became the exact size of message, so it can be negative, but not less than `-priv->frag_proto_min_length`. I can add this check too, I mean `-frag_proto_min_length < ack_data_size <= 0`, WDYT?

Reply:
ok